src/librepgp/stream-sig.cpp

```diff
@@ -1265,36 +1306,78 @@ pgp_signature_t::parse_v4(pgp_packet_body_t &pkt)
     /* hash algorithm */
     halg = (pgp_hash_alg_t) buf[2];
     /* hashed subpackets length */
-    uint16_t splen = read_uint16(&buf[3]);
-    /* hashed subpackets length + 2 bytes of length of unhashed subpackets */
-    if (pkt.left() < (size_t)(splen + 2)) {
+
+    uint8_t
+      splen_buf[PGP_MAX_SIG_HASHED_AREA_SIZE_OCTETS]; // large enough to hold V4 and V6 lengths
+    unsigned splen;
+    size_t   splen_size;
+
+    if (version == PGP_V4) {
```

Comment:
Regarding `pkt.get_len()` : as it is only internal function, it may rely on real conditions, and allow input only of 16 or 32 bits (or has parameter bytes, instead of bits). Anyway, this may be addressed later on, once adding V5 support.